### PR TITLE
[UR] Bump UR and implement changes to bindless image handle types

### PR DESCRIPTION
**Pull request by callumfare**
https://github.com/oneapi-src/unified-runtime/pull/1829

### CODE REVIEW

File: sycl/include/sycl/detail/pi.h [hunk 1]
```diff
@@ -199,9 +199,10 @@
 // 16.56 Replaced piextUSMEnqueueMemset with piextUSMEnqueueFill
 // 16.57 Added mappings to UR launch properties extension
 // (piextEnqueueKernelLaunchCustom)
+// 16.57 Added context parameter to piextMemImageGetInfo
```
Comment (przemektmalon) [inline sycl/include/sycl/detail/pi.h:202]:
```suggestion
// 16.58 Added context parameter to piextMemImageGetInfo
```

File: sycl/plugins/cuda/pi_cuda.cpp [hunk 2]
```diff
@@ -446,11 +446,12 @@ __SYCL_EXPORT pi_result piextMemSampledImageHandleDestroy(
   return pi2ur::piextMemSampledImageHandleDestroy(Context, Device, Handle);
 }
 
-__SYCL_EXPORT pi_result piextMemImageGetInfo(pi_image_mem_handle MemHandle,
+__SYCL_EXPORT pi_result piextMemImageGetInfo(pi_context Context,
+                                             pi_image_mem_handle MemHandle,
```
Comment (przemektmalon) [inline sycl/plugins/cuda/pi_cuda.cpp:450]: `pi.h` has these two parameters as `const`. Though this was already the case before with `pi_image_mem_handle`, so perhaps the disparity doesn't cause problems.

Comment (callumfare) [reply sycl/plugins/cuda/pi_cuda.cpp:450]: I've removed `const` from both, which matches other handle types

**Review by przemektmalon — APPROVED** [review]

**Review by kbenzie — APPROVED** [review]

File: sycl/include/sycl/detail/pi.h [hunk 3]
```diff
@@ -199,9 +199,10 @@
 // 16.56 Replaced piextUSMEnqueueMemset with piextUSMEnqueueFill
 // 16.57 Added mappings to UR launch properties extension
 // (piextEnqueueKernelLaunchCustom)
+// 16.58 Added context parameter to piextMemImageGetInfo
 
 #define _PI_H_VERSION_MAJOR 16
```
Comment (cppchedy) [inline sycl/include/sycl/detail/pi.h:204]: this should be bumped too if there is a change to `pi*` function signature.

Comment (kbenzie) [reply sycl/include/sycl/detail/pi.h]: done

**Review by PietroGhg — APPROVED** [review]
Native CPU LGTM, thank you

**Review by cppchedy — APPROVED** [review]

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers please merge

**Pull request merged**